src/Prima/Pyxis/Form/Autocomplete.elm

```diff
@@ -257,9 +257,13 @@ updateChoices choices (State state) =
 
 {-| Internal.
 -}
-updateOnFilter : State -> State
-updateOnFilter (State state) =
-    State { state | choices = Loading }
+updateOnFilter : ( State, Cmd Msg, Maybe String ) -> ( State, Cmd Msg, Maybe String )
+updateOnFilter ( State state, cmd, filter ) =
+    ( filter
+        |> ME.unwrap (State state) (always (State { state | choices = Loading }))
+    , cmd
+    , filter
+    )
```

Comment:
Questa ti consiglio di riscriverla come: 
```
updateOnFilter : ( State, Cmd Msg, Maybe String ) -> ( State, Cmd Msg, Maybe String )
updateOnFilter ( ((State state) as stateModel), cmd, filter ) =
(filter 
    |> Maybe.map (always (State { state | choices = Loading }))
    |> Maybe.withDefault stateModel
, cmd
, filter
)
```
che mi pare un pochino + semplice da leggere :)

Quando destrutturi o patternmatchi qualcosa in elm puoi fare una cosa del tipo: 

```
type alias Custom = { x : Int }

myFn : Custom -> Custom 
myFn ({x} as custom) = 
   if x > 10 then 
       {custom | x = x + 1}
   else 
       custom
```

Puoi fare la stessa cosa coi costruttori e i loro tags: 


```
type Custom = Custom { x : Int }

myFn : Custom -> Custom 
myFn ((Custom {x}) as custom) = 
   if x > 10 then 
       {custom | x = x + 1}
   else 
       custom
```